controllers/activemqartemis_fsm.go

```diff
@@ -74,6 +77,8 @@ type Namers struct {
 }
 
 type ActiveMQArtemisFSM struct {
+	requestedResources []rtclient.Object
```

Comment:
maybe because of the inline call to fms.Update? I think that is problematic,

Reply:
yes that is because of the security update can happen at any moment, the requiest resources will be accessed by concurrent recociles and will cause contention. Put it the fsm make sure each reconcile will have it's own requested resources. (one example is that if you deployed n broker CRs - i.e. multiple statefulsets and when security applied to them they will be concurrent reconcile calls on the controller)